microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -388,12 +332,15 @@ func (r *GMConnectorReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			graph.Spec.RouterConfig.Config = make(map[string]string)
 		}
 		graph.Spec.RouterConfig.Config["nodes"] = "'" + jsonString + "'"
-		err = reconcileResource(ctx, dynamicClient, graph.Spec.RouterConfig.Name, router_ns, graph.Spec.RouterConfig.ServiceName, &graph.Spec.RouterConfig.Config, nil)
+		//set empty service name, because we don't want to change router service name and deployment name
```

Comment:
1. we need update graph for router even if router already exists. otherwise, router can not get updated graph when user changed their GMC CR.
2. we need add GMC CR name for the router service and deployment to support each CR in the same namespace has its own router service.

Reply:
ok